.github/actions/sonar/action.yml

```
@@ -0,0 +1,253 @@
+          export $ENV_VAR_NAME=$METRIC_VALUE
+        }
+
+        export_measure "new_vulnerabilities" "NEW_VUL"
+        export_measure "new_code_smells" "NEW_CSM"
+        export_measure "new_bugs" "NEW_BUG"
+        export_measure "new_violations" "NEW_VIOLATIONS"
+        export_measure "new_coverage" "NEW_COV"
+        export_measure "new_duplicated_lines_density" "NEW_DUP"
+
+        {
+          echo "## :bar_chart: SonarQube Analysis for ${{ inputs.sonar-project-key }}"
+          echo ""
+          echo "| Metric                      | Value       |"
+          echo "|-----------------------------|-------------|"
+          echo "| **New Vulnerabilities**     | ${NEW_VUL} |"
+          echo "| **New Bugs**                | ${NEW_BUG} |"
+          echo "| **New Code smells**         | ${NEW_CSM} |"
+          echo "| **Coverage on New Code**    | ${NEW_COV}% |"
+          echo "| **Duplication on New Code** | ${NEW_DUP}% |"
+
+          if [[ "${NEW_VIOLATIONS}" != "N/A" && "${NEW_VIOLATIONS}" != "0" ]]; then
+            echo ""
+            echo "<details><summary>Unresolved Issues (click to expand)</summary>"
+            echo ""
+            ISSUES=$(curl -s -u admin:admin \
+              "http://localhost:9000/api/issues/search?componentKeys=${{ inputs.sonar-project-key }}&resolved=false&pullRequest=${{ github.event.pull_request.number }}" | \
+              jq -r '.issues[] | "File: \(.component) Line: \(.line)\n  [\(.rule)] \(.message)\n"')
+            echo "$ISSUES"
+            echo "</details>"
+          fi
+        } > sonar_result.md
+
+        {
+          echo 'COMMENT_BODY<<EOF'
+          cat sonar_result.md
+          echo 'EOF'
+        } >> "$GITHUB_ENV"
+
+    # Find previous SonarQube analysis comment
+    - name: 'Find existing PR comment with SonarQube results'
+      if: github.event_name == 'pull_request' && steps.sonar_metrics.outcome == 'success'
+      id: find_comment
+      uses: peter-evans/find-comment@v3
+      with:
+        issue-number: ${{ github.event.pull_request.number }}
+        body-includes: '## :bar_chart: SonarQube Analysis for ${{ inputs.sonar-project-key }}'
+        token: ${{ inputs.comment-token }}
+
+    # Create or update PR comment with SonarQube results
+    - name: 'Post SonarQube results as PR comment'
+      if: github.event_name == 'pull_request' && steps.sonar_metrics.outcome == 'success'
+      uses: peter-evans/create-or-update-comment@v4
+      with:
+        issue-number: ${{ github.event.pull_request.number }}
+        body: ${{ env.COMMENT_BODY }}
+        comment-id: ${{ steps.find_comment.outputs.comment-id }}
+        edit-mode: replace
+        token: ${{ inputs.comment-token }}
+
+    # Fail the action if there are unresolved issues
+    - name: 'Fail PR if unresolved issues are found'
+      if: github.event_name == 'pull_request' && steps.sonar_metrics.outcome == 'success' && env.NEW_VIOLATIONS != 'N/A' && env.NEW_VIOLATIONS != 0
```

Allow to opt-out sonar failure
```suggestion
      if: >-
        github.event_name == 'pull_request' &&
        steps.sonar_metrics.outcome == 'success' &&
        env.NEW_VIOLATIONS != 'N/A' && env.NEW_VIOLATIONS != 0 &&
        !contains(github.event.pull_request.labels.*.name, 'pr: disable-sonar')
```